pkg/cloudprovider/drift.go

```diff
@@ -71,11 +73,19 @@ func (c *CloudProvider) isAMIDrifted(ctx context.Context, nodeClaim *karpv1.Node
 	if err != nil {
 		return "", fmt.Errorf("getting instanceTypes, %w", err)
 	}
+
+	if _, ok := nodeClaim.Labels[corev1.LabelInstanceTypeStable]; !ok {
+		return "", fmt.Errorf("nodeclaim doesn't have instance type label")
+	}
 	nodeInstanceType, found := lo.Find(instanceTypes, func(instType *cloudprovider.InstanceType) bool {
 		return instType.Name == nodeClaim.Labels[corev1.LabelInstanceTypeStable]
 	})
+	// If we can't find the instance type, that means that we previously thought this was a valid instance type, but no longer is.
+	// This should only happen when the result of the DescribeInstanceTypes API changes over time, and an existing node has an instance type
+	// that should no longer be used.
 	if !found {
-		return "", fmt.Errorf(`finding node instance type "%s"`, nodeClaim.Labels[corev1.LabelInstanceTypeStable])
+		log.FromContext(ctx).V(1).Info(fmt.Sprintf("failed to find node instance type %q, continuing to drift", nodeClaim.Labels[corev1.LabelInstanceTypeStable]))
```

Comment:
This seems like it might be generally applicable to all providers. Why just have this in the AWS provider?

Reply:
I imagine that different providers might have different ways to reason about instance type deprecations. I wouldn't want to be overhanded and assume we know how it should work for everyone